htlcswitch/iterator.go

```diff
@@ -193,6 +193,36 @@ func (r *sphinxHopIterator) ForwardingInstructions() (ForwardingInfo, error) {
 
 		nextHop = lnwire.NewShortChanIDFromInt(cid)
```

Comment:
Is this still needed? I would think that checking `hasNextHop` is enough now?

Reply:
we still need to check that if it `hasNextHop`, that they didn't provide an all zeroes sid

Reply:
Ok. I guess that is the consequence of us using a magic value internally..

Reply:
Indeed 😂